rust-runtime/aws-smithy-runtime-api/src/client/interceptors/context.rs

```diff
@@ -327,6 +336,32 @@ where
     }
 }
 
+/// The result of attempting to rewind a request.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum RewindResult {
```

Comment:
Seems like this should be `non_exhaustive`, or private somehow. Not blocking this PR, but maybe we should move `InterceptorContext` into `aws-smithy-runtime` and have separate customer facing interceptor contexts in the `-api` crate so that all the orchestrator details are well hidden.

Reply:
I'll make it `#[doc(hidden)]` for now. I'm guessing we'll do a reorganization pass before GA.